### ISSUE TITLE
Improvements to Contentlet as JSON

### ISSUE THREAD

**Issue by nollymar** [dotCMS/core#21356]
Testing this [card](https://github.com/dotCMS/core/issues/21011) I found some issues that aren't blockers but I think it is important to fix: 

- [ ] This log message can become too noisy: 
```
dotcms_1         | 16:11:12.604  WARN  business.ContentletJsonAPIImpl - Unable to set field `Image2` with the given value /data/shared/assets/d/d/ddabc075-921d-4cb9-bdac-a45f8d7a1d99/image2/Gyde Calor Heated Mens Vest 2.png.
```

The message above is displayed during deploy using a fresh installation with full starter.

- [ ] When none value is set to a float/integer field, the field remains empty. However, on previous versions default values are used instead (0.0 for floats and 0 for integers)

![Screen Shot 2021-12-03 at 10.00.45 AM.png](https://images.zenhubusercontent.com/5730f592bf93871ccd9ac00c/8ae535f3-222a-49e7-99c7-985fd3e8678b)

![Screen Shot 2021-12-03 at 9.58.47 AM.png](https://images.zenhubusercontent.com/5730f592bf93871ccd9ac00c/f822bdf4-4c03-4535-9b38-49f960870c71)

**Comment by alfredo-dotcms** [comment]
![image](https://user-images.githubusercontent.com/37185433/145850477-8d439846-ab71-4c1a-a355-6c4f460f9abf.png)

looks good!

**Comment by bryanboza** [comment]
Fixed, tested on release-21.12 // Docker // FF
